### ISSUE TITLE
jface High-DPI: FileImageDescriptor's autoscaling strategy unnecessarily restricted to quadratic images

### ISSUE THREAD

**Issue by Dani-Hub** [eclipse-platform/eclipse.platform.ui#375]
When attempting to find an existing file suitable for a higher resolution, [FileImageDescriptor.java#L171](https://github.com/eclipse-platform/eclipse.platform.ui/blob/ed41af6b6d85026dcc987cc1a899dba49d2c29c5/bundles/org.eclipse.jface/src/org/eclipse/jface/resource/FileImageDescriptor.java#L171) uses a regex suitable for the general, rectangular case, but uses the result in a way that only gives the expected result for a quadratic image, where both width and height are equal:
```
int current = Integer.parseInt(matcher.group(1));
int desired = (int) ((zoom / 100d) * current);
String lead = name.substring(0, matcher.start(1));
String tail = name.substring(matcher.end(2));
return lead + desired + "x" + desired + tail; //$NON-NLS-1$
```
Assuming a filename exists of the name "Hula-hup-16x57.png" for a zoom value of 200 the current algorithm will attempt to find a file "Hula-hup-32x32.png". It might even find such a file (Which would be surely not the expected image the user would like to be selected), but the natural expectation would be to look for a file named "Hula-hup-32x114.png" instead.

It would be a low-hanging fruit to make the existing calculation also work for filenames associated with possibly rectangular images as follows
```
int currentWidth = Integer.parseInt(matcher.group(1));
int desiredWidth = (int) ((zoom / 100d) * currentWidth);
int currentHeight = Integer.parseInt(matcher.group(2));
int desiredHeight = (int) ((zoom / 100d) * currentHeight);
String lead = name.substring(0, matcher.start(1));
String tail = name.substring(matcher.end(2));
return lead + desiredWidth + "x" + desiredHeight + tail; //$NON-NLS-1$
```

**Comment by mbooth101** [comment]
Looks like this was fixed by PR #379 right?

**Comment by akurtakov** [comment]
Yes, looks like it.
